ciao-image/datastore/posix.go

```diff
@@ -27,20 +27,23 @@ type Posix struct {
 }
 
 // Write copies an image into the posix filesystem.
-func (p *Posix) Write(ID string, body io.Reader) (int64, error) {
+func (p *Posix) Write(ID string, body io.Reader) error {
 	imageName := path.Join(p.MountPoint, ID)
 	if _, err := os.Stat(imageName); !os.IsNotExist(err) {
-		return 0, fmt.Errorf("image already uploaded with that ID")
+		return fmt.Errorf("image already uploaded with that ID")
 	}
 
 	image, err := os.Create(imageName)
 	if err != nil {
-		return 0, err
+		return err
 	}
 
 	buf := make([]byte, 1<<16)
 
-	return io.CopyBuffer(image, body, buf)
+	_, err = io.CopyBuffer(image, body, buf)
+	defer image.Close()
```

Comment:
As you are writing to image you need to check the return value from Close.  I usually do this by using a named return value, e.g.,

```
func fn() (err error) {
   ....

   defer func() {
     err1 := image.Close()
     if err == nil {
       err = err1
     }
    }()
}

```